types/hapi__joi/hapi__joi-tests.ts

```diff
@@ -1158,6 +1158,13 @@ schema = Joi.symbol().map({
 
 // --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- ---
 
+const rule = Joi.string().case('upper').$_getRule('case');
+if (rule && rule.args) {
+    const direction = rule.args.direction;
```

Comment:
I still don't understand this, maybe another review will do a better job than me — What's the `const direction = rule.args.direction` doing here? It's just a constant declaration and doesn't have any tests/etc. to use its value?

Reply:
That's right. The purpose of this PR is to fix a bug for the typing of `$_getRule`. Current typing implementation returns a type `ExtensionRule`, which is technically incorrect compared with what the javascript is returning. Most of the properties on `ExtensionRule` actually diverge from the [response from](https://github.com/hapijs/joi/blob/b3728a27513f88469022183814a8c4416ee709b1/lib/base.js#L646-L649) `$_getRule`. One example being the test I've added here: 
```typescript
const rule = schema.$_getRule('case');

// PR changes
rule.args            // type of Record<string, any>
rule.args.direction  // works because underlying `args` are an object

// Current release
rule.args            // type of Array<RuleArgs | string>
rule.args.direction  // typescript thinks `args` is an array, so accessing `direction` throws an error
```
This test follows the same pattern this test for terms:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f7ca01985e75715920c20791367e028ed529772c/types/hapi__joi/hapi__joi-tests.ts#L1162-L1163

Let me know if I can clarify further.